lib/rest-adapter.js

```diff
@@ -369,22 +369,41 @@ RestAdapter.errorHandler = function(options) {
         res.statusCode = err.statusCode || err.status || 500;
       }
 
-      debug('Error in %s %s: %s', req.method, req.url, err.stack);
-      var data = {
-        name: err.name,
-        status: res.statusCode,
-        message: err.message || 'An unknown error occurred',
-      };
-
-      for (var prop in err) {
-        data[prop] = err[prop];
+      if (Array.isArray(err)) {
+        var details = err.map(function(it) {
+          var data = generateResponseError(it);
+          delete data.statusCode;
+          return data;
+        });
+
+        err = new Error('Failed with multiple errors, see `details` for more information.');
+        err.details = details;
       }
 
-      data.stack = err.stack;
-      if (process.env.NODE_ENV === 'production' || options.disableStackTrace) {
-        delete data.stack;
+      debug('Error in %s %s: %s', req.method, req.url, err.stack);
```

Comment:
On the second thought, I think it may be a bit surprising that `generateResponseError` prints a debug log as a side-effect. Even you, as an author, got confused.

I am proposing to remove the `debug` statement from `generateResponseError`, and add explicit `debug()` calls in all places where it makes sense. Please check the resulting debug output for both cases (single error, array of errors) to ensure the information printed is useful and easy to understand.